AUTHORS: GROUP 3 (Enhancer - H3K4me1+H3K27ac). State 3
-Lucía de Dios Blázquez
-Sergio Muñoz González
-Alicia Peñaranda Navazo
-Jorge Sastre Domínguez
-Daniel Alejandro Prieto Cebollero


This work has been developed for the subject Transcriptomics, Genomic Regulation and Epigenomics in MSc of Bioninformatics and Computational Biology at Autonomous University of Madrid(UAM)

An epigenomic analysis of the Enhancer (H3K4me1) was performed in monocitytes.
Output files in this project have been gathered using software like PeakAnnotator, HOMER, GREAT or UCSC browser.

TASKS:
Obtain the segments that have the same state in the two replicates of monocytes. 
 
Annotate  the  segments.  As  a  minimum,  the  percentage  of  segments  that  overlap 
with protein-coding genes. 
 
Download  the  DNase  I  peaks  for  monocytes  from  ENCODE  and  calculate  the 
percentage  of  overlap  between  DNaseI-peaks  and  your  working  segments. 
 
Visualize (and show) a region of the genome in the UCSC browser where you see 
at  least  one  of  your  segments  ( the  track  generated uploaded by  ChromHMM)  and  the 
DNaseI in all ENCODE blood cell types. 
 
Search for enriched motifs. 
 
Calculate  the  %  of  overlap  with  hyper-  (Methylation>  0.75)  or 
hypo-methylated (Methylation <0.25) regions in monocytes available in the 
BLUEPRINT portal DCC.

